src/main/scala/io/iohk/praos/domain/ledger/ConsensusResolver.scala

```diff
@@ -1,9 +1,26 @@
-package io.iohk.praos.domain
+package io.iohk.praos.domain.ledger
 
-object ConsensusResolver {
+import io.iohk.praos.domain.{Blockchain, BlockchainState}
 
-  def receiveChain(newChain: Blockchain, state: BlockchainState): BlockchainState = {
-    BlockchainState(state.fullBlockchain, state.receivedChains :+ newChain)
+trait ConsensusResolver {
+  def pickMaxValid(state: BlockchainState): BlockchainState
+}
+
+
+object ConsensusResolver extends ConsensusResolver {
+
+  /**
+    * @return The new BlockchainState, with the current max valid blockchain and without received chains.
+    */
+  def pickMaxValid(state: BlockchainState): BlockchainState = {
```

Comment:
Should we maybe add a `TODO` comment regarding the possible pending validation of the txs of the blocks?

Reply:
:+1: